### PR TITLE
Drops poetry and Makefile

### PR DESCRIPTION
**Pull request by funkyfuture**
when i was trying to install delb in a musl-based environment from source, the build-system found it necessary to install a plethora of dependencies of which one git-related required a `gcc`. i lost my patience and found that a move away from poetry should be done earlier than later. i considered *setuptools* and *hatchling* as build backend and chose the latter because there's already a plugin for *hatch* that allows to employ *mypyc* during package builds. at the same time *hatch* allowed a reasonable way to manage development dependencies. it adds an extra layer of virtual environments to my workflow, it performs well enough to be ignored.

this change deliberately drops version pinning. for the library's dependencies the pinning is a potential security issue and i don't expect any breaking releases (*lxml* is supposed to be removed anyway and *cssselect* might some day also be replaced with a native implementation). if the development dependencies break, be it so, it's always fixable.

i'm opening this PR now in order to get the Github workflows running and for feedback. i'll likely release [cff-from-621](https://github.com/delb-xml/cff-from-621) and amend its employment here.

For clarity the `Makefile` is also removed and the entrypoint discovery is modernized.

### CODE REVIEW

**Comment by ofek** [comment]
Awesome stuff 🎉

File: docs/installation.rst [hunk 1]
```diff
@@ -50,8 +49,43 @@ tools::
...
+interpreter three tools need to be available globally. pipx_ is the recommended
+facilitation to install the Python implemented tools *black* and *hatch*.
+
+just
+~~~~
+
+just_ is a task runner that executes a variety of common *recipes*. This gives a
+list of all available ones::
+
+    …/delb-py $ just -list
```
Comment (funkyfuture) [inline docs/installation.rst:65]:
```suggestion
    …/delb-py $ just --list
```

File: docs/installation.rst [hunk 2]
```diff
@@ -50,8 +49,43 @@ tools::
...
+
+It's recommended to configure the used editors and IDEs to enforce black_'s code
+style, but it can also be applied with::
+
+    …/delb-py $ just black
+
+hatch
+~~~~~
+
+Many other of the *just* recipes rely on hatch_.
```
Comment (funkyfuture) [inline docs/installation.rst:83]:
```suggestion
Several of the *just* recipes rely on hatch_.
```

**Comment by funkyfuture** [comment]
i'm unsure whether i should be creeped out due to your stalking (just out of interest, how did you notice this PR, data mining i assume?) or well-supervised by the facilitator of the employed tool. anyway, thanks for your feedback, it seems that hatch is well documented and usable.

i'm closing this PR and merge a re-arranged set of commits locally.

**Comment by ofek** [comment]
Haha yeah every weekend I search for open PRs & issues to see if I can help

**Pull request closed without merging**